### PR TITLE
New api

### PR DESCRIPTION
**Pull request by tomphp**

### CODE REVIEW

**Comment by tomphp** [comment]
@localheinz would love some thoughts on this new API (docs not yet updated). Specifically:

* Does it read well (i.e. `ConfigureContainer::fromFiles($container, ['config.json]);`)
* Is deprecating `TomPHP\ConfigServiceProvider\ConfigServiceProvider` a good idea or should we just remove it?
* Are the settings name good or could they be improved?

Thanks :-)

File: README.md [hunk 1]
```diff
@@ -23,110 +23,131 @@ $ composer require tomphp/config-service-provider
...
+            ],
+        ],
+    ],
 ];
 
 $container = new Container();
-
-$container->addServiceProvider(ConfigServiceProvider::fromConfig($appConfig));
-$container->addServiceProvider(new DatabaseConnectionProvider());
+ConfigureContainer::fromArray($container, $config);
```
Comment (localheinz) [inline README.md]: @tomphp 

I have been thinking about the naming, and I am not entirely sure, but maybe we can come up with something better, what do you think?

Just wondering, maybe a usage similar to this makes sense:

```php
ContainerConfigurator::for($container)
    ->usingSettings($settings)
    ->usingConfiguration($configuration)
    ->usingConfigurationFiles($files)
    ->apply()
;
```
Is that too much?

A bit like a builder, instead, it doesn't return anything.

Comment (localheinz) [inline README.md]: @tomphp 

The reason I have been thinking about this:

```php
ConfigureContainer::fromArray($container, $config);
```

feels like it is a named constructor that will return an instance of `ConfigureContainer` - but that isn't the case, and to me, `ConfigureContainer` sounds a bit weird (no offense), just unsure about the naming.

Comment (tomphp) [inline README.md]: @localheinz I agree that it sounds a bit weird, it's not a naming style or programming style I'd normally go for. I just wanted to find a way to people to use the library with a bare minimum learning effort.

I like "builder" style one you've suggested, but I worry it makes it easier for users to make mistakes - (i.e. forget to call `apply()` at the end). I'm not against it but I think this should be consider.

What about a different class and method name, i.e. `Configurator::configureContainerFromArray()`?

Or we could ever remove the class all together and just have a function - would that be clearer?

**Pull request merged**